live-ingester/src/integration-test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/ATSTransformResponseHandlerTest.java

```diff
@@ -0,0 +1,244 @@
+        {}
+
+        // then
+        then(ingestContentCommandHandler).should(times(retryIngestion.attempts() + 1)).handle(expectedCommand);
+    }
+
+    @ParameterizedTest
+    @ValueSource(classes = {EmptyRenditionException.class, ResourceNotFoundException.class})
+    void shouldRetryTransformationOnSpecificExceptions(Class<? extends Throwable> exception)
+    {
+        // given
+        String nodeId = "dcc9e89b-c1c0-48a4-814f-98eafab72fdb";
+        String transformedFileId = "e5bcc533-853c-44f1-a02d-3ab1e36a03e6";
+        String targetMimeType = "application/pdf";
+        int retryAttempt = 1;
+
+        TransformResponse transformResponse = new TransformResponse(
+                transformedFileId,
+                new ClientData(
+                        nodeId, targetMimeType, 0),
+                202,
+                "");
+
+        IngestContentCommand expectedCommand = new IngestContentCommand(
+                transformedFileId,
+                nodeId);
+
+        doThrow(exception).when(ingestContentCommandHandler).handle(expectedCommand);
+
+        TransformRequest expectedTransformRequest = new TransformRequest(nodeId, targetMimeType);
+
+        // when
+        try
+        {
+            simulateResponse(transformResponse);
+        }
+        catch (CamelExecutionException ignored)
+        {}
+
+        // then
+        then(ingestContentCommandHandler).should(times(1)).handle(expectedCommand);
+        then(atsTransformRequester).should().requestTransformRetry(expectedTransformRequest, retryAttempt);
+    }
+
+    @ParameterizedTest
+    @ValueSource(classes = {EmptyRenditionException.class, ResourceNotFoundException.class})
+    void shouldNotRetryTransformationIfMaxAttemptsNumberIsExceeded(Class<? extends Throwable> exception)
+    {
+        // given
+        String nodeId = "dcc9e89b-c1c0-48a4-814f-98eafab72fdb";
+        String transformedFileId = "e5bcc533-853c-44f1-a02d-3ab1e36a03e6";
+        String targetMimeType = "application/pdf";
+
+        TransformResponse transformResponse = new TransformResponse(
+                transformedFileId,
+                new ClientData(
+                        nodeId, targetMimeType, retryTransformation.attempts()),
+                202,
+                "");
+
+        IngestContentCommand expectedCommand = new IngestContentCommand(
+                transformedFileId,
+                nodeId);
+
+        doThrow(exception).when(ingestContentCommandHandler).handle(expectedCommand);
+
+        // when
+        try
+        {
+            simulateResponse(transformResponse);
+        }
+        catch (CamelExecutionException ignored)
+        {}
+
+        // then
+        then(ingestContentCommandHandler).should(times(1)).handle(expectedCommand);
+        then(atsTransformRequester).shouldHaveNoInteractions();
+    }
+
+    @SneakyThrows
+    private void simulateResponse(TransformResponse response)
+    {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Can this be a field and then the whole method be replaced by the single call to camel?

In fact do we need to create a new `camelContext.createProducerTemplate()` for each test?

Reply:
tbh I like current version more as it keeps a lot of test implementation details in this method and we don't have configuration mess at the start of the class (+ it costs us nothing in test, it does not have to be super performant and reuse everyting)

Speaking about ProducerTemplate - I have the same feelings as about ObjectMapper but I think it is also worth to mention that I used it in exactly same way we use it in production code (we are creating new ProducerTemplate for each request - maybe it is a mistake and we should create one and reuse it?)
<img width="820" alt="image" src="https://github.com/Alfresco/hxinsight-connector/assets/62990104/416a1e47-ffdd-4a73-ba32-d7a2f96dbf9b">

Reply:
I replaced CamelContext bean with ProducerTemplate bean